### PR TITLE
 [Distributed] Set stdin to devnull before closing it

### PR DESCRIPTION
**Pull request by vtjnash**
Distributed closes and destroys stdin, but some tests attempted to
explicitly use it, leading to test problems. We previously interpreted
this as passing devnull, but this is better to be explicit.

@DilumAluthge @staticfloat

### CODE REVIEW

**Review by staticfloat — APPROVED** [review]
Looks good to me, let's see how CI does!

File: stdlib/REPL/test/TerminalMenus/legacytests/old_radio_menu.jl [hunk 1]
```diff
@@ -36,8 +36,8 @@ TerminalMenus.writeLine(buf, radio_menu, 1, true)
 
 # Test using stdin
 radio_menu = RadioMenu(string.(1:10), warn=false)
-@test simulate_input(3, radio_menu, :down, :down, :enter)
+@test simulate_input(radio_menu, :down, :down, :enter) == 3
```
Comment (vtjnash) [inline stdlib/REPL/test/TerminalMenus/legacytests/old_radio_menu.jl:39]:
```suggestion
@test simulate_input(radio_menu, :down, :down, :enter) == 2
```

File: stdlib/REPL/src/TerminalMenus/util.jl [hunk 2]
```diff
@@ -17,24 +17,24 @@ readbyte(stream::IO=stdin) = read(stream, Char)
...
-readkey(stream::Base.LibuvStream=stdin) = UInt32(_readkey(stream))
-function _readkey(stream::Base.LibuvStream=stdin)
+readkey(stream::IO=stdin) = UInt32(_readkey(stream))
+function _readkey(stream::IO=stdin)
     c = readbyte(stream)
 
     # Escape characters
     if c == '\x1b'
-        stream.buffer.size < 2 && return '\x1b'
+        bytesavailable(stream) < 2 && return '\x1b'
```
Comment (vtjnash) [inline stdlib/REPL/src/TerminalMenus/util.jl:26]:
```suggestion
        bytesavailable(stream) < 1 && return '\x1b'
```

File: stdlib/REPL/src/TerminalMenus/util.jl [hunk 3]
```diff
@@ -17,24 +17,24 @@ readbyte(stream::IO=stdin) = read(stream, Char)
...
         esc_a = readbyte(stream)
         esc_a == 'v' && return PAGE_UP  # M-v
         esc_a == '<' && return HOME_KEY # M-<
         esc_a == '>' && return END_KEY  # M->
 
-        stream.buffer.size < 3 && return '\x1b'
+        bytesavailable(stream) < 3 && return '\x1b'
```
Comment (vtjnash) [inline stdlib/REPL/src/TerminalMenus/util.jl:32]:
```suggestion
        bytesavailable(stream) < 1 && return '\x1b'
```

File: stdlib/REPL/src/TerminalMenus/util.jl [hunk 4]
```diff
@@ -17,24 +17,24 @@ readbyte(stream::IO=stdin) = read(stream, Char)
...
         esc_b = readbyte(stream)
 
         if esc_a == '[' || esc_a == 'O'
             if esc_b >= '0' && esc_b <= '9'
-                stream.buffer.size < 4 && return '\x1b'
+                bytesavailable(stream) < 4 && return '\x1b'
```
Comment (vtjnash) [inline stdlib/REPL/src/TerminalMenus/util.jl:37]:
```suggestion
                bytesavailable(stream) < 1 && return '\x1b'
```

**Comment by staticfloat** [comment]
Buildkite CI looks good to me; `x86_64-apple-darwin` and `x86_64-linux-gnu` both passed.

**Comment by vtjnash** [comment]
Do I need to revert any other PRs to properly test this now?

**Comment by staticfloat** [comment]
Just the `REPL` and `Pkg`, which you've done just now.

**Comment by DilumAluthge** [comment]
Thank you @vtjnash and @staticfloat!

**Comment by vtjnash** [comment]
I don't know why buildbot is broken, but most of buildkite looked good and just the common crashes in Pkg and LibGit2 tests on musl?

**Comment by DilumAluthge** [comment]
@vtjnash Can you take a look at the REPL tests on x86_64-apple-darwin on Buildkite? They are failing.

https://buildkite.com/julialang/julia-master/builds/10862#cffe05fa-9ecb-4eb3-8255-0987bc738687

```
Error in testset REPL:
Test Failed at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-x64-5.0/build/default-macmini-x64-5-0/julialang/julia-master/julia-6552c91511/share/julia/stdlib/v1.9/REPL/test/repl.jl:285
  Expression: s == "\e[0m\n"
   Evaluated: "\r\r\r\e[0K\e[1A\r\e[0K\e[1A\r\e[0K\e[1A\r\e[0K\e[1A\r\e[0K\e[1A\r\e[0K\e[1A\r\e[0K\e[31m\e[1mshell> \e[0m\e[0m\r\e[7C/Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-x64-5.0/build/default-macmini-x64-5-0/julialang/julia-master/julia-6552c91511/bin/julia -Cnative -J/Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-x64-5.0/build/default-macmini-x64-5-0/julialang/julia-master/julia-6552c91511/lib/julia/sys.dylib --depwarn=error --check-bounds=yes -g1 --startup-file=no -e \"println(\\\"HI\\\")\" \n" == "\e[0m\n"
Error in testset REPL:
Test Failed at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-x64-5.0/build/default-macmini-x64-5-0/julialang/julia-master/julia-6552c91511/share/julia/stdlib/v1.9/REPL/test/repl.jl:290
  Expression: fetch(get_stdout) == "HI\n"
   Evaluated: "" == "HI\n"
```

**Comment by vtjnash** [comment]
The rest passed, so I assume it is a flaky test with writing too much or too long of a path or something

**Comment by DilumAluthge** [comment]
I've restarted it; let's see if it passes the second time.

**Comment by DilumAluthge** [comment]
@staticfloat The Buildbot `tester_macos64` is also failing, but the log does not seem to explain why.

**Comment by DilumAluthge** [comment]
@vtjnash I get the same `REPL` failures when I rerun the `x86_64-apple-darwin` job, so I don't think it's a flaky test.

https://buildkite.com/julialang/julia-master/builds/10862#c5ccc5ab-aef0-43d2-a275-b8582bfaf614

**Review by staticfloat — APPROVED** [review]

**Comment by DilumAluthge** [comment]
I'm seeing the known segfaults on x86_64-apple, but x86_64-linux-gnu looks happy.

**Comment by DilumAluthge** [comment]
Actually, Pkg did not run.... ?

**Comment by DilumAluthge** [comment]
Oh we skip Pkg under rr.

**Comment by DilumAluthge** [comment]
Both `Pkg` and `REPL` passed on `x86_64-linux-gnu`.

(Note: we skip `Pkg` when running under `rr`, so you'll need to look at the non-`rr` job.

**Pull request merged**